pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -281,8 +285,15 @@ func (e *eksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 			cluster, err = e.generateAndSetServiceAccount(cluster)
 			if err != nil {
 				var statusErr error
-				cluster, statusErr = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting,
-					fmt.Sprintf("failed to communicate with cluster: %v", err))
+				if strings.Contains(err.Error(), fmt.Sprintf(dialer.WaitForAgentError, cluster.Name)) {
+					// In this case, the API endpoint is private and rancher is waiting for the import cluster command to be run.
+					cluster, statusErr = e.setUnknown(cluster, apimgmtv3.ClusterConditionWaiting, "waiting for cluster agent to be deployed")
+					if statusErr == nil {
+						e.clusterEnqueueAfter(cluster.Name, enqueueTime)
+					}
+					return cluster, statusErr
+				}
+				cluster, statusErr = e.setFalse(cluster, apimgmtv3.ClusterConditionWaiting, fmt.Sprintf("failed to communicate with cluster: %v", err))
```

Comment:
Nit: Since this line didn't actually change we could leave as-is.

Reply:
Reverted.